memory viewer

idle-PB opened the pull request:
Added Decimal Octal Hex views to the table data view, via a button that cycles the mode and a checkbox data to optionally format the output for a data section on copy or save. 
Added ~700 ASM mnemonics, they will also have to be added to the compiler. 
added a Linux-x64.sh build with instructions on how to compile but it's dirty as it requires copying the gtk2 subsystem libs over the gtk3 libs otherwise it will link both and won't run.

fantaisie-software commented on PureBasicDebugger/MemoryViewer.pb:
```diff
@@ -3,23 +3,128 @@
+EndProcedure 
+
+Procedure.s OCT(number.q)
+  Protected oct.s=Space(8*SizeOf(Character))
```
Are you sure you need to multiply with SizeOf(Character) ? Space() already takes NbChar as parameter

idle-PB replied:
#DEBUGGER_GADGET_Memory_ChkformatDataSection needs to be added into DebuggerCommon.pb it got wiped messing around with git which also broke my build. 

no probably don't need the * sizeOf(Character) either in that case, I'll look into it.

fantaisie-software commented on PureBasicDebugger/MemoryViewer.pb:
```diff
@@ -3,23 +3,128 @@
+    number >> 3
+  Next 
+  oct = Trim(oct,"0") 
+  If oct = "        " 
```
The whole IDE use uppercase as first variable name, would be great to continue like that :)

idle-PB replied:
Ok I will try

fantaisie-software commented on PureBasicDebugger/MemoryViewer.pb:
```diff
@@ -43,7 +148,7 @@ Procedure MemoryViewer_Table(*Debugger.DebuggerData, VariableSize, PeekVal.Memor
       AddGadgetColumn(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_List], i+1, Hex(i * VariableSize), width)
-    Next i
+    Next i   
     
```
why the extra space at line end ?

fantaisie-software commented on PureBasicDebugger/MemoryViewer.pb:
```diff
@@ -269,8 +374,9 @@ Procedure MemoryViewer_Unicode(*Debugger.DebuggerData)
 Procedure MemoryViewer_Update(*Debugger.DebuggerData, Action, File) ; 0=display only, 1=display+copy to clipboard, 2=display+save to file
-  
+  Protected datatype.s, OptExportDS=1 
   ViewType = GetGadgetState(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_ViewType])
```
datatype -> DataType

fantaisie-software commented on PureBasicDebugger/MemoryViewer.pb:
```diff
@@ -479,6 +616,7 @@ Procedure MemoryViewerWindowEvents(*Debugger.DebuggerData, EventID)
     TextWidth    = Max(65, GetRequiredWidth(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_Text]))
     
     ButtonHeight = Max(ButtonHeight, GetRequiredHeight(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_ViewType]))
+    dataWidth    = MAX(100, GetRequiredWidth(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_ChkformatDataSection]))
```
DataWidth

fantaisie-software commented on PureBasicDebugger/MemoryViewer.pb:
```diff
@@ -633,6 +777,8 @@ Procedure UpdateMemoryViewerWindow(*Debugger.DebuggerData)
   SetGadgetText(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_Text], Language("Debugger","Range")+":")
   SetGadgetText(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_Display], Language("Debugger","Display"))
+  SetGadgetText(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_Display_DataView],MemoryViewer_GetDataView(MemoryViewTableData))
+  SetGadgetText(*Debugger\Gadgets[#DEBUGGER_GADGET_Memory_ChkformatDataSection],"Data")
```
DEBUGGER_GADGET_Memory_ChkformatDataSection -> DEBUGGER_GADGET_Memory_CheckFormatDataSection

kenmo-pb commented:
I have not tested this yet, but there are some odd things in the diff / Files Changed.

There are some lines that are just whitespace changes (spaces added at end of line?) and some comment typos are reintroduced (is there a mix-up of old and new file versions again?)

idle-PB commented:
git is driving me crazy, the memory viewer wasn't the right file, I just re uploaded it along with debuggercommon so hopefully that will sort it out but I fear, it will only stuff it up more.

fantaisie-software commented:
Don't worry to much idle-PB, we all been experienced hard debut with git haha :). Thanks for resubmitting the PR, I will take a look

idle-PB commented:
as an example  428 in Dec : 654 in Oct : $1AC in Hex 
fpem.u = 428
ShowMemoryViewer(@fpem,2)

and yes maybe a combo box would be better

idle-PB commented:
changed to combo box, see previous comment about how it should look.

fantaisie-software commented:
Did you pushed it ? I can't see the combo. I see what you mean now, but it only works in 'table' mode right ? If we stay in 'Hex View', it has no effect. If so, you should gray it in 'Hex View' and enable it when the view is switched to a table. You could even move the combo next to the view selection as it's directly related.

idle-PB commented:
I committed it but it didn't appear where it should, then I tried to append it but that didn't work out either and then I just added it to a branch memview. I'm still trying to figure out git / github. 

And just now I cocked it up again it appears on my master not the branch!  
I moved it next to the viewtype combo and set it to enabled and disabled based on the viewtype

idle-PB commented:
looks like I'll have to close this pull request and then do another from the memview branch I created.

tajmone commented:
@idle-PB, remember that I've kept a backup of the previous PR (plus my UTF-8 fix):

https://github.com/tajmone/purebasic/tree/idle-PB-fix-memory-viewer

... in case you need it (to recover the files).

idle-PB commented:
ok thanks but can I close this pull request and re do it from the memview branch?

idle-PB commented:
created a new pull from memview branch since I can't see how  to append the changes to this.

tajmone commented:
> created a new pull from memview branch

You may want to close this PR then?
